### ISSUE TITLE
Case-splitting in parameters block incorrectly generates impossible cases

### ISSUE THREAD

**Issue by ohad** [idris-lang/Idris2#1529]
# Steps to Reproduce
```idris
parameters (Str : String)
  foo : (n : Nat) -> Unit
  foo n = ?foo_rhs
```
and case-split (interactively) on `n`.
# Expected Behavior

```idris
parameters (Str : Bool)
  foo : (n : Nat) -> Unit
  foo Z = ?foo_rhs1
  foo (S k) = ?foo_rhs2
```

# Observed Behavior
Generates the code:

```idris
parameters (Str : String)
  foo : (n : Nat) -> Unit
(Str : String) -> foo n impossible
(Str : String) -> foo n impossibl
```

**Comment by CodingCellist** [comment]
Fixed by #1674? : )

**Comment by CodingCellist** [comment]
AFAICT, Idris2 now correctly case-splits on the example above. Please re-open if it's still causing problems  : )
